### ISSUE TITLE
Incorrect missing @throws error in methods that use closures

### ISSUE THREAD

**Issue by mindplay-dk** [squizlabs/PHP_CodeSniffer#790]
The following incorrectly leads to a `missing @throws` inspection:

```php
<?php

class Foo
{
    /**
     * @return Closure
     */
    public function getStuff()
    {
        return function () {
            throw new RuntimeException("bam!");
        };
    }
}
```

The method itself doesn't throw - the closure does.

Throw statements inside closures should not trigger this inspection.

**Comment by aik099** [comment]
Unfortunately there is no way to know, during static file analysis, if that closure isn't called anywhere inside same method. That's why it happens. Same thing happens, when throw happens inside IF statement.

See more at https://github.com/squizlabs/PHP_CodeSniffer/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+throws

**Comment by mindplay-dk** [comment]
This is a false positive by any other name.

Please seem my comments on #738

**Comment by gsherwood** [comment]
I'm pretty much over this one, so I've made the change. So now the default position is "false negative", which seems like the wrong decision for a coding standards checker, but there you go.
